nengo_gui/components/netgraph.py

```diff
@@ -162,9 +162,52 @@ def _reload(self, code=None):
         orphan_components = []
         rebuild_components = []
 
+        # items that are shown in components, but not currently displayed
+        #  in the NetGraph (i.e. things that are inside collapsed
+        #  Networks, but whose values are being shown in a graph)
+        collapsed_items = []
+
         removed_items = list(removed_uids.values())
         for c in self.page.components[:]:
             for item in c.code_python_args(old_default_labels):
+                if item not in self.uids.keys() and item not in collapsed_items:
+
+                    # item is a python string that is an argument to the
+                    # constructor for the Component.  So it could be 'a',
+                    # 'model.ensembles[3]', 'True', or even 'target=a'.
+                    # We need to evaluate this string in the context of the
+                    # locals dictionary and see what object it refers to
+                    # so we can determine whether to rebuild this component.
+                    #
+                    # The following lambda should do this, handling both
+                    # the normal argument case and the keyword argument case.
+                    safe_eval = '(lambda *a, **b: list(a) + b.values())(%s)[0]'
+
+                    # this Component depends on an item inside a collapsed
+                    #  Network, so we need to check if that component has
+                    #  changed or been removed
+                    old_obj = eval(safe_eval % item, old_locals)
+
+                    try:
+                        new_obj = eval(safe_eval % item, self.page.locals)
+                    except:
+                        # the object this Component depends on no longer exists
+                        new_obj = None
+
+                    keep_object = True
+                    if new_obj is None:
```

Comment:
It's not identical.  The code around 128 doesn't need the distinction between removed_items and rebuilt_items, so it doesn't do that.  It'd certainly be possible and a good idea to refactor this stuff in general, such that it'd be possible to have identical code here, but I think that's outside the scope of this bugfix.

That said, the keep_object=True line should be removed...  :)